ducktape/cluster/remoteaccount.py

```diff
@@ -30,7 +61,99 @@ def __init__(self, hostname, user=None, ssh_args=None, ssh_hostname=None, extern
         self.ssh_args = ssh_args
         self.ssh_hostname = ssh_hostname
         self.externally_routable_ip = externally_routable_ip
-        self.logger = logger
+        self._logger = logger
+        self._ssh_config = None
+        self._ssh_client = None
+        self._sftp_client = None
+
+    @property
+    def logger(self):
+        if self._logger:
+            return self._logger
+        else:
+            return logging.getLogger(__name__)
+
+    @logger.setter
+    def logger(self, logger):
+        self._logger = logger
+
+    def _log(self, level, msg, *args, **kwargs):
+        msg = "%s: %s" % (str(self), msg)
+        self.logger.log(level, msg, *args, **kwargs)
+
+    @property
+    def ssh_config(self):
+        if not self._ssh_config:
+            self._ssh_config = self._parse_ssh_opts()
+        return self._ssh_config
+
+    @property
+    def ssh_client(self):
+        if not self._ssh_client:
+            o = self.ssh_config.lookup(self.hostname)
+
+            client = SSHClient()
+            client.set_missing_host_key_policy(AutoAddPolicy())
```

Comment:
Ok, looks like we can effectively ignore missing host key with a new `MissingHostKeyPolicy`